tapir/coop/tests/membership_resignation/test_service.py

```diff
@@ -17,5 +30,209 @@ def setUp(self) -> None:
+                ),
+                share.end_date,
+            )
+
+    def test_updateShiftsAndSharesAndPayOutDay_SharesAndPayOutDayForResignationTypeGiftToCoop_transferredAndSet(
+        self,
+    ):
+        self.login_as_member_office_user()
+        share_owner = ShareOwnerFactory.create(nb_shares=2)
+        resignation = MembershipResignationFactory.create(
+            share_owner=share_owner,
+            resignation_type=MembershipResignation.ResignationType.GIFT_TO_COOP,
+        )
+        MembershipResignationService.update_shifts_and_shares_and_pay_out_day(
+            resignation=resignation,
+        )
+        shares_after_update = resignation.share_owner.share_ownerships
+        for share in shares_after_update.all():
+            self.assertEqual(share.end_date, resignation.cancellation_date)
+
+    def test_updateShiftsAndSharesAndPayOutDay_SharesAndPayOutDayForResignationTypeTransfer_transferredAndSet(
+        self,
+    ):
+        self.login_as_member_office_user()
+        share_owner = ShareOwnerFactory.create(nb_shares=2)
+        resignation = MembershipResignationFactory.create(
+            share_owner=share_owner,
+            resignation_type=MembershipResignation.ResignationType.TRANSFER,
+        )
+        MembershipResignationService.update_shifts_and_shares_and_pay_out_day(
+            resignation=resignation,
+        )
+        shares_after_update = resignation.share_owner.share_ownerships
+        self.assertEqual(
+            ShareOwnership.objects.filter(
+                share_owner=resignation.transferring_shares_to,
+                start_date=resignation.cancellation_date,
+            ).count(),
+            shares_after_update.count(),
+        )
+        for share in shares_after_update.all():
+            self.assertEqual(share.end_date, resignation.cancellation_date)
+
+    def test_updateShifts_shiftsAndShiftAttendance_cancelled(self):
+        tapir_user = TapirUserFactory.create()
+        share_owner = ShareOwnerFactory.create()
+        resignation = MembershipResignationFactory.create(share_owner=share_owner)
+        shift_template = ShiftTemplateFactory.create(
+            start_date=resignation.cancellation_date + datetime.timedelta(days=1),
+        )
+        shift = ShiftFactory.create()
+        ShiftAttendance.objects.create(user=tapir_user, slot=shift.slots.first())
+        ShiftAttendanceTemplate.objects.create(
+            user=tapir_user, slot_template=shift_template.slot_templates.first()
+        )
+        MembershipResignationService.update_shifts(
+            tapir_user=tapir_user, resignation=resignation
+        )
+        self.assertEqual(ShiftAttendanceTemplate.objects.count(), 0)
+        self.assertEqual(
+            ShiftAttendance.objects.get(user=tapir_user).state,
+            ShiftAttendance.State.CANCELLED,
+        )
+
+    def test_updateShifts_shiftsBeforeResignation_notCancelled(self):
```

Comment:
It looks like there is confusion about the `start_date` of `ShiftTemplate`: it's the date from which the `ShiftTemplate` should start generating shifts. So it's not relevant to resignations: no matter what the `ShiftTemplate.start_date` is, if the resigning member is registered to `ShiftTemplate`, they should get unregistered from that `ShiftTemplate`

Reply:
I thought to achieve this topic (https://github.com/SuperCoopBerlin/tapir/pull/569#discussion_r1836994901), with the `start_date` of the `ShiftTemplate`. Is it more the `Shift start_time`?

Reply:
Yes, the only time to check is the time of the `Shift`: shifts that are before the resignation should not get affected.
The `start_date` of the `ShiftTemplate` is irrelevant.